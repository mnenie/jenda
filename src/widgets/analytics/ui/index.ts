import UsersChart from './UsersChart.vue';

export { UsersChart };
