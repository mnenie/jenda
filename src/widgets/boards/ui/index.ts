import TeamBoards from './TeamBoards.vue';
import ActivityСharts from './ActivityСharts.vue';

export { TeamBoards, ActivityСharts };
