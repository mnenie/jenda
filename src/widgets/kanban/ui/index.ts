import KanbanWrapper from './KanbanWrapper.vue';

export { KanbanWrapper };
