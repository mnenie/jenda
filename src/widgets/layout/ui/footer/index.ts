import FooterWelcome from './FooterWelcome.vue';

export { FooterWelcome };
