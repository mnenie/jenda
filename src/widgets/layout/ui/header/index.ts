import HeaderMain from './HeaderMain.vue';
import HeaderWelcome from './HeaderWelcome.vue';

export { HeaderMain, HeaderWelcome };
