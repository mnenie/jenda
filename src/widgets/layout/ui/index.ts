export * from './sidebar';
export * from './header';
