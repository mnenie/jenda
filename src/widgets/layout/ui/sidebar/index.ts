import AppSidebar from './AppSidebar.vue';

export { AppSidebar };
