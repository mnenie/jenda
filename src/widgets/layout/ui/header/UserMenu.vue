<script setup lang="ts">
import { UiDropdown, UiDropdownItem, UiDropdownShortcut } from '@/shared/ui';
import { RouteNames } from '@/shared/config/consts';
import { Menu } from '@/shared/assets/icons';
</script>

<template>
  <UiDropdown>
    <template #trigger>
      <div :class="$style.wrapper">
        <Menu :class="$style.icon" />
      </div>
    </template>
    <template #header>
      <span class="text-sm" style="font-weight: 500">test@gmail.com</span>
    </template>
    <template #content>
      <UiDropdownItem @click="$router.push({ name: RouteNames.welcome })">
        {{ $t('header.user.welcome') }}
        <UiDropdownShortcut>⌘B</UiDropdownShortcut>
      </UiDropdownItem>
      <UiDropdownItem>
        {{ $t('header.user.logout') }}
        <UiDropdownShortcut>⇧⌘Q</UiDropdownShortcut>
      </UiDropdownItem>
    </template>
  </UiDropdown>
</template>

<style module lang="scss">
.wrapper {
  height: 17px;
}

.icon {
  width: 17px;
  height: 17px;
  color: var(--zinc-500);
  cursor: pointer;
}

:global(html.dark) {
  .icon {
    color: var(--zinc-300);
  }
}
</style>
