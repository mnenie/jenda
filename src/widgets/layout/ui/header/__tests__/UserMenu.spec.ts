import { describe, it, expect, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import '@/shared/lib/vitest-utils/cookiesI18n-mock';
import i18n from '@/shared/lib/i18n';
import UserMenu from '../UserMenu.vue';
import { UiDropdownItem } from '@/shared/ui';

const mockRouter = {
  push: vi.fn(),
  beforeEach: vi.fn()
};

describe('tests for UserMenu.vue', () => {
  const wrapper = mount(UserMenu, {
    global: {
      plugins: [i18n],

      mocks: {
        t: (key: string) => {
          const translations: Record<string, string> = {
            'header.user.welcome': 'welcome',
            'header.user.logout': 'logout'
          };
          return translations[key];
        },
        $router: mockRouter
      }
    }
  });

  it('should be render correctly', async () => {
    // TODO: change {name: 'Component'} to Component(instance) for (!all tests) -> name - deprecated
    await wrapper.findComponent({ name: 'UiDropdown' }).vm.handleDropdown();
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should redirect coorectly', async () => {
    const welcomeTrigger = wrapper.find('.content').findAllComponents(UiDropdownItem).at(0);

    await welcomeTrigger.trigger('click');
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'welcome' });
  });
});
