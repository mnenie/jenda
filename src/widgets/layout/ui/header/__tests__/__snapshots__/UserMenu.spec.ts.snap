// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for UserMenu.vue > should be render correctly 1`] = `
"<div class="dropdown">
  <div class="wrapper">
    <div class="wrapper"><svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24" class="icon">
        <path fill="none" stroke="currentColor" stroke-linejoin="round" stroke-width="3.75" d="M12 12h.01v.01H12zm0-7h.01v.01H12zm0 14h.01v.01H12z"></path>
      </svg></div>
  </div>
  <transition-stub name="dropdown" appear="false" persisted="false" css="true">
    <div class="inside">
      <div class="header"><span class="text-sm" style="font-weight: 500;">test@gmail.com</span></div>
      <div class="content">
        <div class="item">Go to Welcome <div class="shortcut">⌘B</div>
        </div>
        <div class="item">Log out <div class="shortcut">⇧⌘Q</div>
        </div>
      </div>
    </div>
  </transition-stub>
</div>"
`;
